pkg/querier/frontend/frontend.go

```diff
@@ -179,46 +180,67 @@ func (f *Frontend) Handler() http.Handler {
 }
 
 func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	// buffer the body for later use in case of slow query
+	var buf bytes.Buffer
+	r.Body = http.MaxBytesReader(w, r.Body, int64(maxBodySize))
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, &buf))
 
 	startTime := time.Now()
 	resp, err := f.roundTripper.RoundTrip(r)
 	queryResponseTime := time.Since(startTime)
 
 	if err != nil {
 		writeError(w, err)
-	} else {
-		hs := w.Header()
-		for h, vs := range resp.Header {
-			hs[h] = vs
-		}
-		w.WriteHeader(resp.StatusCode)
-		io.Copy(w, resp.Body)
+		return
 	}
 
-	// If LogQueriesLongerThan is set to <0 we log every query, if it is set to 0 query logging
-	// is disabled
-	if f.cfg.LogQueriesLongerThan != 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
-		logMessage := []interface{}{
-			"msg", "slow query detected",
-			"method", r.Method,
-			"host", r.Host,
-			"path", r.URL.Path,
-			"time_taken", queryResponseTime.String(),
-		}
+	hs := w.Header()
+	for h, vs := range resp.Header {
+		hs[h] = vs
+	}
 
-		// Ensure the form has been parsed so all the parameters are present
-		err = r.ParseForm()
-		if err != nil {
-			level.Warn(util.WithContext(r.Context(), f.log)).Log("msg", "unable to parse form for request", "err", err)
-		}
+	w.WriteHeader(resp.StatusCode)
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
```

Comment:
We shouldn't call `writeError` if copying body to response fails. `WriteError` cannot possibly succeed at this point, and may generate extra log messages since we try to send another status code. We can either log the message or silently ignore the error as before.

Reply:
Yes, removed error handling and added comment saying why we don't handle it